File: scalechain-proto-codec/src/test/scala/io/scalechain/blockchain/proto/codec/indexparts/RecordLocatorCodecSpec.scala
package io.scalechain.blockchain.proto.codec.indexparts

import io.scalechain.blockchain.proto.RecordLocator
import io.scalechain.blockchain.proto.codec.{PayloadTestSuite, RecordLocatorCodec}
import io.scalechain.util.HexUtil._

class RecordLocatorCodecSpec extends PayloadTestSuite[RecordLocator]  {

  val codec = RecordLocatorCodec.codec

  val payload = bytes(
    """
      0a 00 00 00 00 00 00 00   c8 00 00 00
    """)

  val message = RecordLocator(
    offset = 10,
    size = 200
  )
}

File: scalechain-proto-codec/src/test/kotlin/io/scalechain/blockchain/proto/codec/indexparts/RecordLocatorCodecSpec.kt
package io.scalechain.blockchain.proto.codec.indexparts


import io.kotlintest.KTestJUnitRunner
import io.scalechain.blockchain.proto.codec.PayloadTestSuite
import io.scalechain.util.HexUtil.bytes
import org.junit.runner.RunWith

import io.scalechain.blockchain.proto.RecordLocator
import io.scalechain.blockchain.proto.codec.RecordLocatorCodec

@RunWith(KTestJUnitRunner::class)
class RecordLocatorCodecSpec : PayloadTestSuite<RecordLocator>()  {

  override val codec = RecordLocatorCodec

  override val payload = bytes(
    """
      0a 00 00 00 00 00 00 00   c8 00 00 00
    """)

  override val message = RecordLocator(
    offset = 10,
    size = 200
  )
}
